content/references/protocol-reference/ledger-data/ledger-object-types/amm.md

```diff
@@ -0,0 +1,138 @@
+labels:
+  - AMM
+status: not_enabled
+---
+# AMM
+[[Source]](https://github.com/gregtatcam/rippled/blob/c1e4bfb08bcc9f187d794a71d653003a6148dc68/src/ripple/protocol/impl/LedgerFormats.cpp#L265-L275 "Source")
+<!-- TODO: Update source link to merged version when available -->
+
+{% include '_snippets/amm-disclaimer.md' %}
+
+The `AMM` object type describes a single Automated Market Maker (AMM) instance.
+
+
+## Example AMM JSON
+
+```json
+{
+  "AMMAccount": "rLcJnZS6M9DyZ23g3GPvtyuYgT5QFwWXaN",
+  "AMMToken": {
+    "Token1": {
+      "TokenCurrency": "0000000000000000000000000000000000000000",
+      "TokenIssuer": "0000000000000000000000000000000000000000"
+    },
+    "Token2": {
+      "TokenCurrency": "0000000000000000000000005453540000000000",
+      "TokenIssuer": "F2F97C4301C80D60F86653A319AA7F302C70B83B"
+    }
+  },
+  "AuctionSlot": {
+    "Account": "rJVUeRqDFNs2xqA7ncVE6ZoAhPUoaJJSQm",
+    "AuthAccounts": [
+      {
+        "AuthAccount": {
+          "Account": "rMKXGCbJ5d8LbrqthdG46q3f969MVK2Qeg"
+        }
+      },
+      {
+        "AuthAccount": {
+          "Account": "rBepJuTLFJt3WmtLXYAxSjtBWAeQxVbncv"
+        }
+      }
+    ],
+    "DiscountedFee": 0,
+    "Price": {
+      "currency": "D717347ECCF3E2B373546F02AA016D7ABA13C8F7",
+      "issuer": "rLcJnZS6M9DyZ23g3GPvtyuYgT5QFwWXaN",
+      "value": "0.7115124735378855"
+    },
+    "TimeStamp": 719029222
+  },
+  "Flags": 0,
+  "LPTokenBalance": {
+    "currency": "D717347ECCF3E2B373546F02AA016D7ABA13C8F7",
+    "issuer": "rLcJnZS6M9DyZ23g3GPvtyuYgT5QFwWXaN",
+    "value": "71150.53584131501"
+  },
+  "LedgerEntryType": "AMM",
+  "TradingFee": 601,
+  "VoteSlots": [
+    {
+      "VoteEntry": {
+        "Account": "rJVUeRqDFNs2xqA7ncVE6ZoAhPUoaJJSQm",
+        "FeeVal": 600,
+        "VoteWeight": 100000
+      }
+    }
+  ],
+  "index": "4C2D6EBC856DE96E1F4859B140CDE495FD878A86E165712F46FF4F2C1D6F16C1"
+}
+```
+
+## AMM Fields
+
+The `AMM` object has the following fields:
+
+| Field            | JSON Type           | [Internal Type][] | Required? | Description  |
+|:-----------------|:--------------------|:------------------|:-------------|
+| `AMMToken`       | Object              | STObject          | Yes       | Defines the two assets this AMM trades, as an [AMMToken Object](#ammtoken-object). |
+| `AccountRootID`  | String              | AccountID         | Yes       | The address of the special account that holds this AMM's assets. |
+| `AuctionSlot`    | Object              | STObject          | No        | Details of the current owner of the auction slot, as an [Auction Slot object](#auction-slot-object). |
+| `LPTokenBalance` | [Currency Amount][] | Amount            | Yes       | The total outstanding balance of liquidity provider tokens from this AMM instance. The holders of these tokens can vote on the AMM's trading fee in proportion to their holdings, or redeem the tokens for a share of the AMM's assets. |
```

Comment:
This phrasing excludes the fact that LP token holders are earning trading fees while they remain in the pool, in addition to their voting rights. 

When they redeem the LP tokens, they are redeeming their original contribution plus trading fees earned.